### DISCUSSION TITLE
add  `#[serde(with = "time::serde::iso8601")]` attribute to TimeDateTimeWithTimeZone when generating entity

### DISCUSSION THREAD

**Discussion by xiayulu** [SeaQL/sea-orm#1873] [Ideas]
My sea orm features:

``` toml
sea-orm = { version = "0.12.2", features = [
    "sqlx-postgres",
    "runtime-tokio-rustls",
    "macros",
    "with-time",
    "with-json",
    "with-uuid",
] }
```

My generated model:

```rust
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, Serialize)]
#[sea_orm(table_name = "post")]
#[serde(rename_all = "camelCase")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub title: String,
    pub created_at: TimeDateTimeWithTimeZone,
    pub expired_at: TimeDateTimeWithTimeZone,
    pub user_id: i32,
}
```

It's strange when I get serialized datetime as Vec<i32>:

```json
{
        "id": 2643,
        "title": "Sea Orm tutorial",
        "createdAt": [
            2023,
            268,
            4,
            35,
            31,
            514929000,
            0,
            0,
            0
        ],
        "updatedAt": [
            2023,
            268,
            4,
            35,
            31,
            514929000,
            0,
            0,
            0
        ],
        "isDeleted": false,
        "userId": 1
    },
```
Better 

```json
{
        "id": 13701,
        "title": "Sea Orm tutorial",
        "createdAt": "+002023-09-25T06:31:10.614642000Z",
        "updatedAt": "+002023-09-25T06:31:10.614642000Z",
        "isDeleted": false,
        "userId": 1
    },
```

**Comment by rgreinho** [comment]
Facing the same exact problem, so I am posted here to give a bump to this discussion.

I am using seaorm 0.12.14 with the `time` library. In my models I have the created_at/updated_at columns defined as `TimestampWithTimeZone`. 

When I retrieve the models, the `TimestampWithTimeZone` columns are return as arrays with a strange format, like `\"created_at\":[2024,75,16,24,39,144518000,0,0,0]}` but I was expecting an ISO8601, something more  like `\"created_at\":\"+002024-03-15T16:26:38.178125000Z\"` that my frontend can consume directly.

My work around currently is to generate my entities with `sea-orm-cli generate entity -o "entity/src/entities" --with-serde both --date-time-crate time`, then to manually edit the models everywhere I use a `TimestampWithTimeZone` and add this serde attribute: `#[serde(with = "time::serde::iso8601")]`. However this approach does not scale.

Is there a way to have seaorm automatically add the serde attribute for the TimestampWithTimeZone fields?
